### PR TITLE
chore: do not leak on clicks

### LINKED ISSUE #22115: [BUG] Simple click leaks the element
```js
test('should not leak', async ({ page }) => {
  for (let i = 0; i < 100; ++i) {
    await page.evaluate((i) => {
      const element = document.createElement('button');
      element.textContent = 'BUTTON ' + i;
      document.body.appendChild(element);
    }, i);
    await page.click('button');
    await page.evaluate(() => {
      document.body.textContent = '';
    });
  }
});
```

### PR DESCRIPTION
**Pull request by pavelfeldman**
Fixes #22115

### CODE REVIEW

File: packages/playwright-core/src/server/dom.ts [hunk 1]
```diff
@@ -455,6 +455,7 @@ export class ElementHandle<T extends Node = Node> extends js.JSHandle<T> {
         // Do not await here, just in case the renderer is stuck (e.g. on alert)
         // and we won't be able to cleanup.
         hitTargetInterceptionHandle!.evaluate(h => h.stop()).catch(e => {});
+        hitTargetInterceptionHandle!.dispose();
```
Comment (yury-s) [inline packages/playwright-core/src/server/dom.ts:458]: Does it mean we'll call a disposed handle at line 474 below?

File: tests/page/page-leaks.spec.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+ */
+
+import { MultiMap } from '../../packages/playwright-core/lib/utils/multimap';
+import { test, expect } from './pageTest';
+
+function leakedJSHandles(): string {
+  const map = new MultiMap();
+  for (const [h, e] of (globalThis as any).leakedJSHandles.entries()) {
+    const name = `[${h.worldName()}] ${h.preview()}`;
+    if (name === '[main] UtilityScript' || '[utility] UtilityScript' || '[main] InjectedScript' || '[utility] InjectedScript')
```
Comment (yury-s) [inline tests/page/page-leaks.spec.ts:24]: [...].includes(name)

File: tests/page/page-leaks.spec.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { MultiMap } from '../../packages/playwright-core/lib/utils/multimap';
+import { test, expect } from './pageTest';
+
+function leakedJSHandles(): string {
+  const map = new MultiMap();
+  for (const [h, e] of (globalThis as any).leakedJSHandles.entries()) {
```
Comment (dgozman) [inline tests/page/page-leaks.spec.ts:22]: Let's cast to the server's JSHandle type here, for future code refactorings.

**Review by dgozman — APPROVED** [review]

File: packages/playwright-core/src/server/javascript.ts [hunk 4]
```diff
@@ -64,9 +65,11 @@ export class ExecutionContext extends SdkObject {
   private _delegate: ExecutionContextDelegate;
   private _utilityScriptPromise: Promise<JSHandle> | undefined;
   private _contextDestroyedRace = new ScopedRace();
+  readonly worldName: string;
```
Comment (yury-s) [inline packages/playwright-core/src/server/javascript.ts:68]: why is it not types.World?

File: packages/playwright-core/src/server/dom.ts (hunk 1 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/dom.ts:458]: Line 454 is saying this code is exiting.

File: packages/playwright-core/src/server/javascript.ts (hunk 4 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/javascript.ts:68]: Because it is wider than that, I'll rename

**Review by yury-s — APPROVED** [review]

**Pull request merged**